### PR TITLE
Entry accessibility

### PR DESCRIPTION
**Pull request by nyee**
Adds tests to check entry accessibility for kinetics families.

A major addition is the requirement of unimolecular families to adhere to strict guidelines with regards to labelling and allowed subgraphs:

1. All ends groups should have exactly the same labels as their respective root
2. All backbone groups should have all labels of attached end groups
3. All backbones should have end groups which are exactly the same as the root of each subtree
4. All backbones must label all atoms on the shortest path between end groups.

These requirements, in addition to standardizing the reaction families, allow merging groups for testing accessibility inside unimolecular families.

### CODE REVIEW

File: rmgpy/molecule/group.py [hunk 1]
```diff
@@ -1760,7 +1827,7 @@ def mergeOverlappingBenzeneRings(ring1, ring2, od):
                 if x ==0: lastAtom = ring[-1]
                 else: lastAtom = mergedRingDict[index][-1]
                 #add a new atom to the ring and the group
-                newAtom = copyGroup. createAndConnectAtom(['Cb'], lastAtom, [1.5])
```
Comment (KEHANG) [inline rmgpy/molecule/group.py]: How come this kind of error merged in before? we need unittests for this method

File: rmgpy/molecule/group.py [hunk 2]
```diff
@@ -1769,49 +1836,164 @@ def mergeOverlappingBenzeneRings(ring1, ring2, od):
...
     def makeSampleMolecule(self):
         """
         Returns: A sample class :Molecule: from the group
         """
 
-        # print self
-        # print self.atoms
+        modifiedGroup = self.copy(deep = True)
+        # print "initial"
+        # print self.toAdjacencyList()
```
Comment (KEHANG) [inline rmgpy/molecule/group.py]: please remove these debug printouts

File: rmgpy/molecule/parserTest.py [hunk 3]
```diff
@@ -2,18 +2,66 @@
...
+        mol = fromInChI(Molecule(), inchi)
+        self.assertTrue(mol.isIsomorphic(self.methane))
+
+        #Test that atomtypes that rely on lone pairs for identity are typed correctly
+        inchi = "InChI=1S/CH5N/c1-2/h2H2,1H3"
+        mol = fromInChI(Molecule(), inchi)
+        self.assertEquals(mol.atoms[1].atomType, atomTypes['N3s'] )
+
+    #Commented out because current implementation of SMARTS is broken
+    # def test_fromSMARTS(self):
```
Comment (KEHANG) [inline rmgpy/molecule/parserTest.py]: use working in progress decorator?

File: rmgpy/molecule/graph.pyx [hunk 4]
```diff
@@ -321,13 +321,41 @@ cdef class Graph:
                     other.addEdge(edge)
         return other
 
+    cpdef dict copyAndMap(self):
```
Comment (KEHANG) [inline rmgpy/molecule/graph.pyx]: add unittest?

File: rmgpy/data/kinetics/family.py [hunk 5]
```diff
@@ -2182,3 +2182,57 @@ def getTopLevelGroups(self, root):
 
         return groupList
 
+    def mergeGroups(self, backbone, end):
```
Comment (nyee) [inline rmgpy/data/kinetics/family.py]: I actually discussed this briefly with somebody in the group. The other argument is that merging groups currently no context outside of families.

I think its fine either way and made a commit moving this to group if you prefer.

File: testing/databaseTest.py [hunk 6]
```diff
@@ -851,16 +850,27 @@ def general_checkSampleDescendsToGroup(self, group_name, group):
...
         for entryName, entry in group.entries.iteritems():
             print entryName
-            if isinstance(entry.item, Group):
-                # print entryName
-                sampleMolecule = entry.item.makeSampleMolecule()
-                atoms = sampleMolecule.getLabeledAtoms()
-                match = group.descendTree(sampleMolecule, atoms, strict=True)
+            try:
+                if isinstance(entry.item, Group):
+                    # print entryName
```
Comment (nyee) [inline testing/databaseTest.py]: Done and removed a few others too

**Pull request merged**